packages/core/src/routes/connector.ts

```diff
@@ -193,7 +193,15 @@ export default function connectorRoutes<T extends AuthedRouter>(router: T) {
         validateConfig(config);
       }
 
-      await updateConnector({ set: body, where: { id }, jsonbMode: 'replace' });
+      // Once created, target can not be modified.
+      const { target, ...otherMetadata } = body.metadata ?? {};
+      const updates = body.metadata ? { ...body, metadata: otherMetadata } : body;
```

Comment:
should we throw an error if the target occurs in the body instead of silently filtering it?